src/models/target/target_utils.jl

```diff
@@ -192,15 +192,36 @@ function ComponentVector2Vector(cv::ComponentVector)
 end
 
 """
-    function create_interpolation(A::Matrix; n_interp_half::Int)
+    create_interpolation(A::Vector; n_interp_half::Int, dilation_factor=1.0) -> Vector{Float64}
 
-Function to create an intepolation for AD computation combining uniform and quantiles.
+Construct a one-dimensional interpolation grid from the data in `A`, combining
+uniformly spaced and quantile-based sampling points.
+
+This hybrid interpolation grid provides both coverage of the entire range
+of values and higher resolution in regions where `A` has dense data,
+making it useful for interpolation or machine learning applications
+that need balanced sampling.
+
+# Arguments
+- `A::Vector`: Input data vector (typically containing positive values).
+- `n_interp_half::Int`: Number of points used for both the uniform and quantile-based 
+  subsets of the interpolation grid.
+- `dilation_factor::Real=1.0`: Optional multiplier applied to `maximum(A)` to slightly 
+  extend the grid beyond the data range (useful to avoid extrapolation issues).
+
+# Returns
+A sorted, unique vector of interpolation nodes combining:
+- `n_interp_half` uniformly spaced values between `0` and `dilation_factor * maximum(A)`
+- `n_interp_half` quantile-based values computed from the positive entries of `A`
 """
-function create_interpolation(A::Matrix; n_interp_half::Int)
-    A_interp_unif = LinRange(0.0, maximum(A), n_interp_half) |> collect
+function create_interpolation(A::Vector; n_interp_half::Int, dilation_factor = 1.0)
+    A_interp_unif = LinRange(0.0, dilation_factor * maximum(A), n_interp_half) |> collect
     pos_values = A[A .> 0.0]
     if length(pos_values) > 0
-        A_interp_quantiles = quantile!(A[A .> 0.0], LinRange(0.0, 1.0, n_interp_half))
+        # We remove the 0.0 and 1.0 quantile since these are included already in the
+        # uniform interpolation
+        quantile_range = LinRange(0.0, 1.0, n_interp_half + 2)[begin + 1: end - 1]
+        A_interp_quantiles = quantile!(A[A .> 0.0], quantile_range)
         A_interp = vcat(A_interp_unif, A_interp_quantiles)
```

Comment:
Is the interpolation gonna work if the two vectors contain redundant values?
```suggestion
        A_interp = unique(vcat(A_interp_unif, A_interp_quantiles))
```